src/__tests__/logic_dynamic.js

```diff
@@ -0,0 +1,571 @@
+        <ConnectedDogComponent id={8} />
+        <ConnectedDogComponent id={12} />
+      </div>
+    </Provider>
+  )
+
+  expect(store.getState()).toEqual({
+    kea: {},
+    scenes: {
+      people: { 8: { name: 'chirpy' }, 12: { name: 'chirpy' } },
+      dogs: {
+        8: { name: 'wufus', ownerChanged: false },
+        12: { name: 'wufus', ownerChanged: false },
+      },
+    },
+  })
+
+  const dogComponent8 = wrapper.find('DogComponent').findWhere(n => n.prop('id') === 8).node
+  dogComponent8.actions.updateOwner('frank')
+
+  expect(store.getState()).toEqual({
+    kea: {},
+    scenes: {
+      people: { 8: { name: 'frank8' }, 12: { name: 'chirpy' } },
+      dogs: {
+        8: { name: 'wufus', ownerChanged: true },
+        12: { name: 'wufus', ownerChanged: false },
+      },
+    },
+  })
+
+  const personComponent12 = wrapper.find('PersonComponent').findWhere(n => n.prop('id') === 12).node
+  personComponent12.actions.updateName('bart')
+
+  expect(store.getState()).toEqual({
+    kea: {},
+    scenes: {
+      people: { 8: { name: 'frank8' }, 12: { name: 'bart12' } },
+      dogs: { 8: { name: 'wufus', ownerChanged: true }, 12: { name: 'wufus', ownerChanged: true } },
+    },
+  })
+
+  wrapper.unmount()
+})
+
+test('dynamic connected component updates external store', () => {
+  const store = getStore({
+    paths: ['kea', 'scenes', 'dogOwners'],
+    preloadedState: { dogOwners: [] },
+  })
+
+  const personLogic = kea({
+    key: props => props.id,
+    path: key => ['scenes', 'people', key],
+    actions: ({ constants }) => ({
+      updateName: name => ({ name }),
+    }),
+    reducers: ({ actions, key, props, constants }) => ({
+      name: [
+        'chirpy',
+        PropTypes.string,
+        {
+          [actions.updateName]: (state, payload) => {
+            return payload.key === key ? payload.name + payload.key : state
+          },
+        },
+      ],
+    }),
+  })
+  const ConnectedComponent = personLogic(PersonComponent)
+
+  const dogLogic = kea({
+    connect: {
+      props: [
+        personLogic.withKey(props => props.id),
+        ['name as owner'],
+        store => store,
+        ['dogOwners'],
+      ],
+      actions: [personLogic.withKey(props => props.id), ['updateName as updateOwner']],
+    },
+    key: props => props.id,
+    path: key => ['scenes', 'dogs', key],
+    actions: ({ constants }) => ({
+      updateName: name => ({ name }),
+    }),
+    reducers: ({ actions, key, props, constants }) => ({
+      name: [
+        'wufus',
+        PropTypes.string,
+        {
+          [actions.updateName]: (state, payload) => {
+            return payload.key === key ? payload.name + payload.key : state
+          },
+        },
+      ],
+      ownerChanged: [
+        false,
+        PropTypes.boolean,
+        {
+          [actions.updateOwner]: (state, payload) => {
+            return payload.key === key ? true : state
+          },
+        },
+      ],
+      dogOwners: [
```

Comment:
this currently doesn't work, my intention was to update the connected prop setup above:

```
connect: {
  props: [
    //...
    store => store, ['dogOwners'],
  ],
```

dunno if this would be in scope for kea or not, since that branch of the store is outside the control of kea...  any recommendation on how to update it?

Reply:
I think I know this could be solved, by using `path` the logic can be mounted wherever you want.